File: src/plugin/req/req-scan.ts
/*
 * Copyright (c) TIKI Inc.
 * MIT license. See LICENSE file in root directory.
 */

import * as uuid from 'uuid';

import type { Req } from './req';

/**
 * A scan request to the plugin
 */
export class ReqScan implements Req {
  requestId: string;
  dayCutOff?: number;

  constructor(dayCutOff?: number) {
    this.requestId = uuid.v4();
    this.dayCutOff = dayCutOff;
  }
}

File: ios/Plugin/Plugin/Req/ReqScan.swift
/*
 * Copyright (c) TIKI Inc.
 * MIT license. See LICENSE file in the root directory.
 */


import Foundation
import Capacitor

public struct ReqScan {
    let daysCutOff: Int?
    let requestId: String
    
    init(data: CAPPluginCall) {
        self.daysCutOff = data.getInt("dayCutOff", 7)
        self.requestId = data.getString("requestId", "")
    }
}
